cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinder.java

```diff
@@ -61,17 +76,35 @@
       }
     }
     if (!topicsWithBadReplicationFactor.isEmpty()) {
-      return Collections.singleton(createTopicReplicationFactorAnomaly(topicsWithBadReplicationFactor,
+      Map<Boolean, Set<String>> topicsByFixability = new HashMap<>(2);
+      List<ConfigResource> topicResourcesToCheck = new ArrayList<>(topicsWithBadReplicationFactor.size());
+      topicsWithBadReplicationFactor.forEach(t -> topicResourcesToCheck.add(new ConfigResource(ConfigResource.Type.TOPIC, t)));
+      for (Map.Entry<ConfigResource, KafkaFuture<Config>> entry: _adminClient.describeConfigs(topicResourcesToCheck).values().entrySet()) {
```

Comment:
Ideally, we should adopt a solution that avoids calling `describeConfigs` every time.
Can we cache this once, and call again only when needed? -- e.g. when new topics are added, if the cache is invalidated after a predefined timeout, or if the user explicitly invalidates the cache via admin endpoint maybe?